usersession/client/client.go

```diff
@@ -307,6 +312,59 @@ func (client *Client) ServicesStop(ctx context.Context, services []string) (stop
 	return stopFailures, err
 }
 
+func (client *Client) ServicesRestart(ctx context.Context, services []string) (restartFailures []ServiceFailure, err error) {
+	restartFailures, _, err = client.serviceControlCall(ctx, "restart", services)
+	return restartFailures, err
+}
+
+func (client *Client) ServicesReloadOrRestart(ctx context.Context, services []string) (restartFailures []ServiceFailure, err error) {
+	restartFailures, _, err = client.serviceControlCall(ctx, "reload-or-restart", services)
+	return restartFailures, err
+}
+
+// UserServiceUnitStatus corresponds to the systemd.UnitStatus in order to be able
+// to reliable transport the data we need surrounding the members we are interested in
+// from systemd.UnitStatus
+type UserServiceUnitStatus struct {
+	Daemon           string   `json:"daemon"`
+	Id               string   `json:"id"`
+	Name             string   `json:"name"`
+	Names            []string `json:"names"`
+	Enabled          bool     `json:"enabled"`
+	Active           bool     `json:"active"`
+	Installed        bool     `json:"installed"`
+	NeedDaemonReload bool     `json:"needs-reload"`
+}
+
+func (client *Client) ServiceStatus(ctx context.Context, services []string) (map[int][]UserServiceUnitStatus, error) {
+	q := make(url.Values)
+	q.Add("services", strings.Join(services, ","))
+
+	responses, err := client.doMany(ctx, "GET", "/v1/service-status", q, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	info := make(map[int][]UserServiceUnitStatus)
+	for _, resp := range responses {
+		if resp.err != nil {
+			if err == nil {
+				err = resp.err
```

Comment:
shouldn't we return these errors by uid instead of return just the first? also should we distinguish service-status errors from other errors?

Reply:
I think this makes more sense when compared to the rest of the functions. So I updated the code to actually return the exact errors in this case, and to more reflect the other functions in the file. It does distinguish service-status errors from client errors now.